packages/libs/lambda-at-edge/src/build.ts

```diff
@@ -348,6 +352,17 @@ class Builder {
       defaultBuildManifest.publicFiles["/" + pf] = pf;
     });
 
+    // Support trailing slash: https://nextjs.org/docs/api-reference/next.config.js/trailing-slash
+    const nextConfigPath = path.join(this.nextConfigDir, "next.config.js");
+
+    if (await fse.pathExists(nextConfigPath)) {
+      const nextConfig = await import(nextConfigPath);
```

Comment:
@dphang Not all next.js configs would work with this.next.config.js can export an object `{ ... }` , a Promise, or a function. Take a look at https://github.com/serverless-nextjs/serverless-next.js/blob/master/packages/libs/lambda-at-edge/src/lib/createServerlessConfig.ts#L10

Reply:
Ah I see, thanks! I'll update this.

Reply:
I updated the code to handle `next.config.js` exporting a function. However I could not return a Promise as when doing Next.js build, it gets an error: https://github.com/vercel/next.js/blob/master/errors/promise-in-next-config.md. Here's their PR from Feb 2019: https://github.com/vercel/next.js/pull/6476. I don't think a Promise export is supported by Next.js?

Reply:
Also did not see any existing tests where `next.config.js` exported a Promise, so not sure if this was ever actually supported/tested. @danielcondemarin do you have any example?

Reply:
The code example I sent actually I took from https://github.com/vercel/vercel/blob/master/packages/now-next/src/create-serverless-config.ts. Maybe it used to be supported at some point and can be removed. 

Since I took the code from next's repo I only covered a few base cases (https://github.com/serverless-nextjs/serverless-next.js/tree/master/packages/libs/lambda-at-edge/tests/integration) but not a Promise.

Reply:
I see, makes sense! Thanks.